File: src/server-java/test/echopoint/BorderLayoutTest.java
package echopoint;

import static org.junit.Assert.assertEquals;
import nextapp.echo.app.Alignment;
import nextapp.echo.app.Color;
import nextapp.echo.app.Component;
import nextapp.echo.app.Extent;
import nextapp.echo.app.layout.TableLayoutData;

import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;

/**
 * <p>&copy; Copyright 2009 <a href='http://sptci.com/' target='_top'>Sans
 * Pareil Technologies, Inc.</a></p>
 *
 * @author Rakesh Vidyadharan 2009-04-02
 * @version $Id: BorderLayoutTest.java,v 1.1.1.1 2010-04-01 09:47:30 perxi Exp $
 */
public class BorderLayoutTest extends AbstractTest<BorderLayout>
{
  @BeforeClass
  public static void init()
  {
    set( new BorderLayout() );
  }

  @Test
  public void width()
  {
    final Extent width = new Extent( 500 );
    getComponent().setWidth( width );
    assertEquals( "Ensure width set", width, getComponent().getWidth() );
  }

  @Test
  public void north()
  {
    final HtmlLabel label = new HtmlLabel();
    label.setText( "<b>This is the north content.  Should stretch across area.</b>" );

    final TableLayoutData layout = new TableLayoutData();
    layout.setAlignment( Alignment.ALIGN_CENTER );
    layout.setBackground( new Color( 0x0000ff ) );
    label.setLayoutData( layout );

    getComponent().add( label, BorderLayout.Region.north );
    assertEquals( "Ensure component added to north", label,
        getComponent().getComponent( BorderLayout.Region.north ) );
  }

  @Test
  public void west()
  {
    final HtmlLabel label = new HtmlLabel();
    label.setText( "<b>West content</b>" );

    final TableLayoutData layout = new TableLayoutData();
    layout.setBackground( new Color( 0x00ff00 ) );
    label.setLayoutData( layout );

    getComponent().add( label, BorderLayout.Region.west );
    assertEquals( "Ensure component added to west", label,
        getComponent().getComponent( BorderLayout.Region.west ) );
  }

  @Test
  public void centre()
  {
    final HtmlLabel label = new HtmlLabel();
    label.setText( "<b>This is the centre content.  Should fill most of central area.</b>" );

    final TableLayoutData layout = new TableLayoutData();
    layout.setAlignment( Alignment.ALIGN_CENTER );
    layout.setBackground( new Color( 0xa1a1a1 ) );
    label.setLayoutData( layout );

    getComponent().add( label, BorderLayout.Region.center );
    assertEquals( "Ensure component added to centre", label,
        getComponent().getComponent( BorderLayout.Region.center ) );
  }

  @Test
  public void east()
  {
    final HtmlLabel label = new HtmlLabel();
    label.setText( "<b>East content</b>" );

    final TableLayoutData layout = new TableLayoutData();
    layout.setBackground( new Color( 0xff0000 ) );
    label.setLayoutData( layout );

    getComponent().add( label, BorderLayout.Region.east );
    assertEquals( "Ensure component added to east", label,
        getComponent().getComponent( BorderLayout.Region.east ) );
  }

  @Test
  public void south()
  {
    final HtmlLabel label = new HtmlLabel();
    label.setText( "<b>This is the south content.  Should stretch across area.</b>" );

    final TableLayoutData layout = new TableLayoutData();
    layout.setAlignment( Alignment.ALIGN_CENTER );
    layout.setBackground( new Color( 0x2f2f4f ) );
    label.setLayoutData( layout );

    getComponent().add( label, BorderLayout.Region.south );
    assertEquals( "Ensure component added to south", label,
        getComponent().getComponent( BorderLayout.Region.south ) );
  }

  @AfterClass
  public static void finish()
  {
    final Component content = Application.getContent().getTestArea();
    content.removeAll();
    content.add( get() );
  }
}

File: src/client/test/app/BorderLayoutTest.js
/**
 * A test class for the echopoint.BorderLayout client-side component.
 * Displays two container components, one full and the other partial.
 *
 * @author Rakesh 2009-03-31
 * @version $Id: BorderLayoutTest.js,v 1.1.1.1 2010-04-01 09:47:33 perxi Exp $
 */
echopoint.test.BorderLayoutTest = Core.extend(
{
  $construct: function( testArea )
  {
    testArea.add( this._createFull() );
    testArea.add( this._createCentre() );
  },

  _createFull: function()
  {
    var layout = new echopoint.BorderLayout(
    {
      renderId: "echopointUnitTestBorderLayoutFull",
      styleName: "Default"
    } );

    var layoutData = { alignment: "center", background: "#0000ff" };

    var north = new echopoint.HtmlLabel(
    {
      renderId: "echopointUnitTestBoderLayoutFullNorth",
      layoutData: layoutData,
      text: "<b>This is the north content.  Should stretch across area.</b>"
    } );
    layout.addToRegion( north, echopoint.BorderLayout.NORTH );

    layoutData = { background: "#00ff00" };
    var west = new echopoint.HtmlLabel(
    {
      renderId: "echopointUnitTestBoderLayoutFullWest",
      layoutData: layoutData,
      text: "<b>West region</b>"
    } );
    layout.addToRegion( west, echopoint.BorderLayout.WEST );

    layoutData = { alignment: "center", background: "#a1a1a1" };
    var centre = new echopoint.HtmlLabel(
    {
      renderId: "echopointUnitTestBoderLayoutFullCentre",
      layoutData: layoutData,
      text: "<b>This is the centre content.  Should fill most of central area.</b>"
    } );
    layout.addToRegion( centre, echopoint.BorderLayout.CENTER );

    layoutData = { background: "#ff0000" };
    var east = new echopoint.HtmlLabel(
    {
      renderId: "echopointUnitTestBoderLayoutFullEast",
      layoutData: layoutData,
      text: "<b>East region</b>"
    } );
    layout.addToRegion( east, echopoint.BorderLayout.EAST );

    layoutData = { alignment: "center", background: "#2f2f4f" };
    var south = new echopoint.HtmlLabel(
    {
      renderId: "echopointUnitTestBoderLayoutFullSouth",
      layoutData: layoutData,
      text: "<b>This is the south content.  Should stretch across area.</b>"
    } );
    layout.addToRegion( south, echopoint.BorderLayout.SOUTH );

    try
    {
      layout.addToRegion( south, echopoint.BorderLayout.SOUTH );
      document.writeln( "Illegal child exception should have been thrown!" );
    }
    catch ( e )
    {
    }

    return layout;
  },

  _createCentre: function()
  {
    var centre = new echopoint.HtmlLabel(
    {
      renderId: "echopointUnitTestBoderLayoutPartialCentre",
      text: "<b>This is the centre content.  Should fill most of central area.</b>"
    } );

    var layout = new echopoint.BorderLayout(
    {
      renderId: "echopointUnitTestBorderLayoutPartial",
      styleName: "Default"
    } );
    layout.addToRegion( centre );

    return layout;
  }
} );
